engine/frontend/src/component/ServiceInfo.js

```diff
@@ -30,32 +33,43 @@ const ServiceInfo = () => {
     const params = useParams()
     const {name: enclaveName, uuid: serviceUuid} = params;
 
-    useEffect(() => {
-        let stream;
-        const ctrl = new AbortController();
-        const fetchLogs = async () => {
-            stream = await getServiceLogs(ctrl, enclaveName, serviceUuid, appData.apiHost);
+    const startServiceLogCollector = (ctrl) => {
+        let stream
+        const fetchServiceLogs = async () => {
+            stream = await getServiceLogs(ctrl, enclaveName, serviceUuid, appData.apiHost, DEFAULT_SHOULD_FOLLOW_LOGS, DEFAULT_NUM_LINES);
             try {
                 for await (const res of stream) {
                     const log = res["serviceLogsByServiceUuid"][serviceUuid]["line"][0]
-                    setLogs(logs => [...logs, log])
+                    if (log !== "") {
+                        setLogs(logs => [...logs, log])
+                    } 
                 }
             } catch (ex) {
-                console.log("aborted stream!")
+                console.log("Abort Initial Log Stream!")
             }
         }
-        fetchLogs()
+
+        fetchServiceLogs()
+    }
+    
+    useEffect(() => {
+
+        const ctrl = new AbortController();
+        startServiceLogCollector(ctrl)
+
         return () => {
+            setLogs([])
             ctrl.abort()
-            setLogs([]);
         };
     }, [serviceUuid])
 
     const handleServiceClick = (service) => {
-        const fullPath = `/enclaves/${enclaveName}/services/${service.uuid}`
+        const fullPath = `/enclaves/${enclaveName}/services/${service.serviceUuid}`
         navigate(fullPath, {state: {services, selected: service}})
     }
 
+
+
```

Comment:
if you are removing these lines, can you also remove the func below that is not in use (not your work)

Reply:
are you sure; we use that for `CodeEditor`. @adschwartz

Reply:
yeah you can just make an anonymous function, but not a big deal. Feel free to skip this.